src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CSharpInstructionDecoder.cs

```diff
@@ -86,7 +87,29 @@ internal override void AppendFullName(StringBuilder builder, PEMethodSymbol meth
             }
         }
 
-        internal override PEMethodSymbol GetMethod(DkmClrInstructionAddress instructionAddress)
+        internal override MethodSymbol ConstructMethod(MethodSymbol method, ImmutableArray<TypeParameterSymbol> typeParameters, ImmutableArray<TypeSymbol> typeArguments)
+        {
+            var methodArity = method.Arity;
+            var methodArgumentStartIndex = typeParameters.Length - methodArity;
+            var typeMap = new TypeMap(
+                ImmutableArray.CreateRange(typeParameters, 0, methodArgumentStartIndex, t => t),
+                ImmutableArray.CreateRange(typeArguments, 0, methodArgumentStartIndex, t => t));
+            var substitutedType = typeMap.SubstituteNamedType(method.ContainingType);
+            method = method.AsMember(substitutedType);
+            if (methodArity > 0)
+            {
+                method = method.Construct(ImmutableArray.CreateRange(typeArguments, methodArgumentStartIndex, methodArity, t => t));
+            }
+            return method;
+        }
+
+        internal override MethodSymbol GetMethod(DkmClrInstructionAddress instructionAddress)
+        {
+            TypeNameDecoder<PEModuleSymbol, TypeSymbol> typeNameDecoder;
+            return GetMethod(instructionAddress, out typeNameDecoder);
```

Comment:
Refactored so that GetCompilation, GetMethod and GetTypeNameDecoder are separate operations...